src/coreclr/jit/compiler.cpp

```diff
@@ -4928,6 +4921,15 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
 
             if (doLoopHoisting)
             {
+                if (doBranchOpt)
+                {
+                    // bbNum quirk: RBO uses the current BB epoch size to determine if loop hoisting created blocks.
+                    // Ensure the BB epoch is up to date before running loop hoisting to avoid false positives.
+                    // TODO-bbNum: Either replace this check with a breadcrumb like 'fgModified',
+                    // or relax jump threading's prerequisites.
+                    EnsureBasicBlockEpoch();
```

Comment:
On second thought, we should be able to get rid of RBO's dependency on block epoch altogether. Let me do that first.